core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationStatusServiceSpec.js

```diff
@@ -0,0 +1,195 @@
+        },
+        Third: {
+          content: {
+            en: {
+              filename: 'filename3.mp3',
+              file_size_bytes: 100000,
+              needs_update: false
+            }}}};
+        return ContentIdsToAudioTranslationsObjectFactory.
+          createFromBackendDict(citatDict[stateName]);
+      },
+      getInteractionIdMemento: function(stateName) {
+        var interactionIds = {First: 'MultipleChoiceInput',
+          Second: 'MultipleChoiceInput',
+          Third: 'EndExploration'};
+        return interactionIds[stateName];
+      }
+    });
+    $provide.constant('INTERACTION_SPECS', {
+      MultipleChoiceInput: {
+        is_linear: false,
+        is_terminal: false
+      },
+      EndExploration: {
+        is_linear: false,
+        is_terminal: false
+      }
+    });
+  }));
+
+  describe('Translation status service', function() {
+    var tss = null;
+    beforeEach(inject(function($injector) {
+      tss = $injector.get('TranslationStatusService');
+      ess = $injector.get('ExplorationStatesService');
+      ContentIdsToAudioTranslationsObjectFactory = $injector.get(
+        'ContentIdsToAudioTranslationsObjectFactory');
+      StateContentIdsToAudioTranslationsService = $injector.get(
+        'StateContentIdsToAudioTranslationsService');
+
+      // To call _computeAllStatesStatus() function of
+      // TranslationStatusService, so that the status of all states is
+      // computed for the other dependent functions to work.
+      tss.getAllStateStatusColors();
+    }));
+
+    it('should return a correct list of state names for which ' +
+      'audio needs update', function() {
+      var statesNeedingAudioUpdate = tss.getAllStatesNeedUpdatewarning();
+      expect(Object.keys(statesNeedingAudioUpdate).length).toBe(1);
+      expect(statesNeedingAudioUpdate.First.
+        indexOf('Audio needs update!')).toBe(0);
+    });
+
+    it('should return a correct count of audio translations required ' +
+      'in an exploration', function() {
+      var explorationAudioRequiredCount = tss.
+        getExplorationAudioRequiredCount();
+      expect(explorationAudioRequiredCount).toBe(9);
+    });
+
+    it('should return a correct count of audio translations not available ' +
+      'in an exploration', function() {
+      var explorationAudioNotAvailableCount = tss.
+        getExplorationAudioNotAvailableCount();
+      expect(explorationAudioNotAvailableCount).toBe(6);
+    });
+
+    it('should correctly return an object contaning status colors of all ' +
+      'states in exploration', function() {
+      var stateWiseStatusColor = tss.getAllStateStatusColors();
+      expect(stateWiseStatusColor.First).toBe('#E9B330');
+      expect(stateWiseStatusColor.Second).toBe('#D14836');
+      expect(stateWiseStatusColor.Third).toBe('#16A765');
+    });
+
+    it('should return correct status color for active state', function() {
+      StateContentIdsToAudioTranslationsService.init(
```

Comment:
can we define this stateNames in a constant? (maybe through indexing the value of `getStateNames`? (not sure!))

Reply:
done

Reply:
@rakshitkumarcse, I don't see any changes related to this comment? are you sure it's done?

Reply:
@DubeySandeep oh sorry. I misunderstood your comment earlier. Making the change.

Reply:
done

Reply:
I cannot see any changes related to this, let me know if it's not clear, I'll try to explain this.

Reply:
Oh, I thought you were expecting [this](https://github.com/oppia/oppia/blob/92d9c4dd3358b7606ec84b18015ab5574f60e868/core/templates/dev/head/pages/exploration_editor/translation_tab/TranslationStatusServiceSpec.js#L145)
Is it something else that you are expecting?

Reply:
@DubeySandeep Does it look good now? Defined the constant STATES for state names.